x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -79,61 +97,250 @@ export class AlertsClient {
+      if (!result?.body.hits.hits.every((hit) => isValidAlert(hit._source))) {
+        const errorMessage = `Invalid alert found with id of "${id}" or with query "${query}" and operation ${operation}`;
+        this.logger.error(errorMessage);
+        throw Boom.badData(errorMessage);
       }
 
-      if (!isValidAlert(result.body.hits.hits[0]._source)) {
-        const errorMessage = `Unable to retrieve alert details for alert with id of "${id}".`;
-        this.logger.debug(errorMessage);
-        throw new Error(errorMessage);
-      }
+      result?.body.hits.hits.map((item) =>
+        this.auditLogger?.log(
+          alertAuditEvent({
+            action: operationAlertAuditActionMap[operation],
+            id: item._id,
+            outcome: 'unknown',
+          })
+        )
+      );
 
-      return {
-        ...result.body.hits.hits[0]._source,
-        _version: encodeHitVersion(result.body.hits.hits[0]),
-      };
+      return result.body;
     } catch (error) {
-      const errorMessage = `Unable to retrieve alert with id of "${id}".`;
-      this.logger.debug(errorMessage);
-      throw error;
+      const errorMessage = `Unable to retrieve alert details for alert with id of "${id}" or with query "${query}" and operation ${operation} \nError: ${error}`;
+      this.logger.error(errorMessage);
+      throw Boom.notFound(errorMessage);
+    }
+  }
+
+  /**
+   * When an update by ids is requested, do a multi-get, ensure authz and audit alerts, then execute bulk update
+   * @param param0
+   * @returns
+   */
+  private async fetchAlertAuditOperate({
+    ids,
+    status,
+    indexName,
+    operation,
+  }: {
+    ids: string[];
+    status: 'open' | 'closed';
+    indexName: string;
+    operation: ReadOperations | WriteOperations;
+  }) {
+    try {
+      const mgetRes = await this.esClient.mget<ParsedTechnicalFields>({
+        index: indexName,
+        body: {
+          ids,
+        },
+      });
+      await Promise.all(
+        mgetRes.body.docs.map((item) => {
+          if (
+            item._source != null &&
+            item._source[RULE_ID] != null &&
+            item._source[OWNER] != null
+          ) {
+            return this.authorization.ensureAuthorized({
+              ruleTypeId: item._source[RULE_ID],
+              consumer: item._source[OWNER],
+              operation,
+              entity: AlertingAuthorizationEntity.Alert,
+            });
+          }
+        })
+      );
+
+      const bulkUpdateRequest = mgetRes.body.docs.flatMap((item) => [
+        {
+          update: {
+            _index: item._index,
+            _id: item._id,
+          },
+        },
+        {
+          doc: { [ALERT_STATUS]: status },
+        },
+      ]);
+
+      const bulkUpdateResponse = await this.esClient.bulk({
+        body: bulkUpdateRequest,
+      });
+      return bulkUpdateResponse;
+    } catch (exc) {
+      this.logger.error(`error in fetchAlertAuthzAlertOperateAlert ${exc}`);
```

Comment:
Is this supposed to be:
```suggestion
      this.logger.error(`error in fetchAlertAuditOperate ${exc}`);
```